cmd/create/cluster/cmd.go

```diff
@@ -1917,6 +1943,113 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	isSharedVPC := false
+	privateHostedZoneID := strings.Trim(args.privateHostedZoneID, " \t")
+	privateHostedZoneRoleARN := strings.Trim(args.privateHostedZoneRoleARN, " \t")
+	baseDomain := strings.Trim(args.baseDomain, " \t")
+	if privateHostedZoneID != "" ||
+		privateHostedZoneRoleARN != "" ||
+		baseDomain != "" {
+		if !isSTS {
+			r.Reporter.Errorf("Installing a cluster into a shared VPC is only supported for STS clusters")
+			os.Exit(1)
+		}
+		if len(subnetIDs) == 0 {
+			r.Reporter.Errorf("Installing a cluster into a shared VPC is only supported for BYO VPC clusters")
+			os.Exit(1)
+		}
+		if isHostedCP {
+			r.Reporter.Errorf("Installing a cluster into a shared VPC is only supported for classic ROSA clusters")
+			os.Exit(1)
+		}
+		isSharedVPC = true
+	}
+	if (privateHostedZoneID == "" ||
```

Comment:
Per Bala's comment here https://docs.google.com/document/d/1f97Xwube-KYscLXQN3SnOMTH3IeK5JAfseJk_vEnlT0/edit?disco=AAAA1ZP5dRU we can identify if the subnets belong to a shared VPC and prompt the interactive questions accordingly.
I think the assumption is that 95% of customers won't use the shared VPC at first, and it can confuse them, therefore, we want to expose it only if they actively shared subnets and chose them.

If possible we should defer the interactive mode handling to a separate PR, and concentrate in this release on enabling the flow via flags for the MVP. Also, we should finalize with BU how is it going to look like for interactive.

Reply:
i agree with that, we should find out first how people are consuming this to better judge how we want to prompt them with out causing confusion. Even the wording shared VPC is confusing. 

Lets capture this in an improvements epic